metron-analytics/metron-profiler-client/src/main/java/org/apache/metron/profiler/client/stellar/GetProfile.java

```diff
@@ -102,94 +90,108 @@
-    //Optional arguments
-    @SuppressWarnings("unchecked")
-    List<Object> groups = null;
-    Map configOverridesMap = null;
-    if (args.size() < 4) {
-      // no optional args, so default 'groups' and configOverridesMap remains null.
-      groups = new ArrayList<>(0);
-    }
-    else if (args.get(3) instanceof List) {
-      // correct extensible usage
-      groups = getArg(3, List.class, args);
-      if (args.size() >= 5) {
-        configOverridesMap = getArg(4, Map.class, args);
-        if (configOverridesMap.isEmpty()) configOverridesMap = null;
-      }
-    }
-    else {
-      // Deprecated "varargs" style usage for groups_list
-      // configOverridesMap cannot be specified so it remains null.
-      groups = getGroupsArg(3, args);
-    }
+    List<ProfilePeriod> periods = getArg(2, List.class, args);
 
-    Map<String, Object> effectiveConfig = getEffectiveConfig(context, configOverridesMap);
-    Object defaultValue = null;
-    //lazily create new profiler client if needed
-    if (client == null || !cachedConfigMap.equals(effectiveConfig)) {
-      RowKeyBuilder rowKeyBuilder = getRowKeyBuilder(effectiveConfig);
-      ColumnBuilder columnBuilder = getColumnBuilder(effectiveConfig);
-      HTableInterface table = getTable(effectiveConfig);
-      long periodDuration = getPeriodDurationInMillis(effectiveConfig);
-      client = new HBaseProfilerClient(table, rowKeyBuilder, columnBuilder, periodDuration);
-      cachedConfigMap = effectiveConfig;
-    }
-    if(cachedConfigMap != null) {
-      defaultValue = ProfilerClientConfig.PROFILER_DEFAULT_VALUE.get(cachedConfigMap);
+    // optional arguments
+    List<Object> groups = getGroups(args);
+    Map<String, Object> overrides = getOverrides(args);
+
+    // lazily create new profiler client if needed
+    Map<String, Object> effectiveConfig = getEffectiveConfig(context, overrides);
+    if (profilerClient == null || !lastKnownGlobals.equals(effectiveConfig)) {
+      profilerClient = profilerClientFactory.create(effectiveConfig);
+      lastKnownGlobals = effectiveConfig;
     }
 
-    List<ProfileMeasurement> measurements = client.fetch(Object.class, profile, entity, groups,
-            periods.orElse(new ArrayList<>(0)), Optional.ofNullable(defaultValue));
+    // is there a default value?
+    Optional<Object> defaultValue = Optional.empty();
+    if(effectiveConfig != null) {
+      defaultValue = Optional.ofNullable(PROFILER_DEFAULT_VALUE.get(effectiveConfig));
+    }
 
     // return only the value of each profile measurement
+    List<ProfileMeasurement> measurements = profilerClient.fetch(Object.class, profile, entity, groups, periods, defaultValue);
     List<Object> values = new ArrayList<>();
     for(ProfileMeasurement m: measurements) {
       values.add(m.getProfileValue());
     }
+
     return values;
   }
 
+  private Map<String, Object> getOverrides(List<Object> args) {
```

Comment:
Yes, I can try.  

I wasn't sure how much I should change here versus not, since this isn't directly necessary for the PR.  I felt at least factoring out the logic into separate methods, but keeping the exact same logic, improved legibility a ton.

Reply:
I hear you - I think this is one instance where it's definitely worth the burden to improve the readability since you've already touched those lines of code refactoring anyways. Naming constants is a pretty common standard practice, versus other refactoring that could be considered stylistic/personal choices on readability, etc.

Reply:
Fixed. The latest commit has this change.  Your suggestion definitely improves things.